dvc/repo/metrics/show.py

```diff
@@ -188,10 +188,16 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
         if not o.metric:
             continue
 
+        # NOTE this case assumes that metric has been modified with
```

Comment:
this case also assumes that `typ` is not defined, not only that metric info is provided via DVC-file